packages/block-library/src/navigation-submenu/index.php

```diff
@@ -239,6 +239,10 @@ function render_block_core_navigation_submenu( $attributes, $content, $block ) {
 			$inner_blocks_html .= $inner_block->render();
 		}
 
+		if ( strpos( $inner_blocks_html, 'current-menu-item' ) ) {
+			$html = str_replace( 'wp-block-navigation-item__content', 'wp-block-navigation-item__content current-menu-ancestor', $html );
```

Comment:
@adamziel Would this be a good opportunity to test out the new tag processor class you've been working on? Seems like a nice and self contained test case.

Reply:
Sounds good indeed! Here's what it could look like:

```php
$p = new WP_HTML_Tag_Processort( $inner_blocks_html );
while ( $p->next_tag( [ 'class_name' => 'wp-block-navigation-item__content' ] ) ) {
    $p->add_class('current-menu-ancestor');
}
$inner_blocks_html = $p->get_updated_html();
```

LMK if that works!

Reply:
Updated to use `WP_HTML_Tag_Processor` 👍